### PR TITLE
Fix issue where DICOM client does not send requests when async ops invoked is zero (= unlimited)

### LINKED ISSUE #1597: DICOM Client does not send requests when Async Ops Invoked = 0 (Meaning unlimited requests are allowed)
**Describe the bug**
Fo-dicom 5.1.0 gets stuck in a loop during CFind request to a fo-dicom 5.0.3 CFind SCP server. It does not happen with all servers, e.g. with Orthanc everything works fine. Downgrading the client to 5.0.3 makes it work again.

The loop never exits and the message "[DBG] (FellowOakDicom.Network) Sending 1 requests" is logged repeatedly (~4000 messages each second).

**To Reproduce**
CFind request code in client:
```
        public async Task<IList<SeriesInfo>> FindSeriesAsync(string studyUid)

        {

            var client = clientFactory.Create(server.Fqdn, server.Port, false, networkConfig.LocalAeTitle, server.AeTitle);

            client.NegotiateAsyncOps();

            var request = CreateSeriesCFindRequest(studyUid);

            await client.AddRequestAsync(request);

            await client.SendAsync(cancellationToken, DicomClientCancellationMode.ImmediatelyAbortAssociation);

 

            return series;

        }

 

        private DicomCFindRequest CreateSeriesCFindRequest(string studyUid)

        {

            var request = DicomCFindRequest.CreateSeriesQuery(studyUid);

            request.OnResponseReceived += (req, response) =>

            {

                Log.Debug($"Received CFind response: {response.Status}");

                if (response.HasDataset)

                {

                    var seriesInfo = GetSeriesInfo(response.Dataset);

                    lock (series)

                    {

                        series.Add(seriesInfo);

                    }

                }

            };

 

            return request;

        }
```

**Expected behavior**
With 5.0.3 the correct response is received. 

**Screenshots or test DICOM files**
Client log:
```
[DBG] (FellowOakDicom.Network) [IDLE] --> [CONNECTING]

[DBG] (FellowOakDicom.Network) [CONNECTING] --> [REQUESTING ASSOCIATION]

[DBG] (FellowOakDicom.Network) Sending association request from CLIENT to TESTSERVER

[INF] (FellowOakDicom.Network) TESTSERVER -> Association request:

Calling AE Title:       CLIENT

Called AE Title:        TESTSERVER

Remote host:            localhost

Remote port:            5555

Implementation Class:   Implementation Class UID [1.3.6.1.4.1.30071.8]

Implementation Version: fo-dicom 5.1.0

Maximum PDU Length:     262144

Async Ops Invoked:      0

Async Ops Performed:    0

Presentation Contexts:  1

  Presentation Context:  1 [Proposed]

       Abstract Syntax:  Study Root Query/Retrieve Information Model - FIND

       Transfer Syntax:  Implicit VR Little Endian: Default Transfer Syntax for DICOM

[INF] (FellowOakDicom.Network) TESTSERVER <- Association accept:

Calling AE Title:       CLIENT

Called AE Title:        TESTSERVER

Remote host:            localhost

Remote port:            5555

Implementation Class:   Unknown [1.3.6.1.4.1.30071.8]

Implementation Version: fo-dicom 5.0.3

Maximum PDU Length:     262144

Async Ops Invoked:      0

Async Ops Performed:    0

Presentation Contexts:  1

  Presentation Context:  1 [Accept]

       Abstract Syntax:  Study Root Query/Retrieve Information Model - FIND

       Transfer Syntax:  Implicit VR Little Endian: Default Transfer Syntax for DICOM

[DBG] (FellowOakDicom.Network) Association request from CLIENT to TESTSERVER has been accepted

[DBG] (FellowOakDicom.Network) [REQUESTING ASSOCIATION] --> [SENDING REQUESTS]

[DBG] (FellowOakDicom.Network) Sending 1 requests

[DBG] (FellowOakDicom.Network) Sending 1 requests

[DBG] (FellowOakDicom.Network) Sending 1 requests
... repeated forever
```

Testserver log:
```
[INFO] CLIENT <- Association request:

Calling AE Title:       CLIENT

Called AE Title:        TESTSERVER

Remote host:            127.0.0.1

Remote port:            59975

Implementation Class:   Implementation Class UID [1.3.6.1.4.1.30071.8]

Implementation Version: fo-dicom 5.1.0

Maximum PDU Length:     262144

Async Ops Invoked:      0

Async Ops Performed:    0

Presentation Contexts:  1

  Presentation Context:  1 [Proposed]

       Abstract Syntax:  Study Root Query/Retrieve Information Model - FIND

       Transfer Syntax:  Implicit VR Little Endian: Default Transfer Syntax for DICOM

[INFO] CLIENT -> Association accept:

Calling AE Title:       CLIENT

Called AE Title:        TESTSERVER

Remote host:            127.0.0.1

Remote port:            59975

Implementation Class:   Implementation Class UID [1.3.6.1.4.1.30071.8]

Implementation Version: fo-dicom 5.1.0

Maximum PDU Length:     262144

Async Ops Invoked:      0

Async Ops Performed:    0

Presentation Contexts:  1

  Presentation Context:  1 [Accept]

       Abstract Syntax:  Study Root Query/Retrieve Information Model - FIND

       Transfer Syntax:  Implicit VR Little Endian: Default Transfer Syntax for DICOM
```

**Environment**
Fellow Oak DICOM version: 5.1.0
OS: Windows 11 Pro
Platform: .NET 7

### PR DESCRIPTION
**Pull request by amoerie**
Fixes #1597 

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
See title

### CODE REVIEW

File: Tests/FO-DICOM.Tests/Bugs/GH1597.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+                    { OnResponseReceived = (req, res) => Interlocked.Increment(ref counter) };
+                await client.AddRequestAsync(request).ConfigureAwait(false);
+            }
+
+            await client.SendAsync().ConfigureAwait(false);
+
+            Assert.Equal(numberOfRequests, counter);
+        }
+    }
+}
```
Comment (mrbean-bremen) [inline Tests/FO-DICOM.Tests/Bugs/GH1597.cs]: I would expect this under the normal client tests, as this tests a standard use case and not some special case revealed in the bug.

Comment (amoerie) [reply Tests/FO-DICOM.Tests/Bugs/GH1597.cs]: Good point. I wasn't too sure. Moved!

**Review by mrbean-bremen — APPROVED** [review]

**Comment by amoerie** [comment]
@gofal This is a bit of a nasty one. There is an easy workaround (use a value like '100' when calling NegotiateAsyncOps) but I'm not sure people will know what to google when they encounter this bug.

This might be a good reason to not wait too long with another release.
You might be irritated with me that such a bug crept through, but all I can say is that I tested in all kinds of ways, but never with dicomClient.NegotiateAsyncOps(). I just never use it that way.
The only lucky aspect here is that most PACSes I'm aware of will not accept Async Ops Invoked 0 and will return 1 or some low number. In those cases, everything works. I think the problem is mostly sending to a fo-dicom SCP.

**Comment by mrbean-bremen** [comment]
Only if you don't do anything, you don't do anything wrong. I think it was to be expected after such a large change that some bug crept in. 
I'm used to make patch releases a couple of days after major releases, and the users are happy if such things are addressed rapidly. Hopefully, a patch release is not that much of work (as the work will be at @gofal's end). I personally would have no problem with an instant patch release. There can always be more patch releases should other regressions be found.

**Comment by gofal** [comment]
sure, i will add an release today or tomorrow. I will set the version 5.1.0 to deprecated, so that users will automatically take 5.1.1

**Review by gofal — APPROVED** [review]

**Pull request merged**